base/sparse/sparsematrix.jl

```diff
@@ -120,7 +120,103 @@ column. In conjunction with [`nonzeros`](@ref) and
 """
 nzrange(S::SparseMatrixCSC, col::Integer) = S.colptr[col]:(S.colptr[col+1]-1)
 
+
+module SparseArrayValidity
+    @enum(_SparseArrayValidity,
+        VALID,
+        COLPTR_LENGTH,
+        COLPTR_FIRST_VAL,
+        NZVAL_LENGTH,
+        COLPTR_SORTED,
+        ROWVAL_LENGTH,
+        ROWVAL_RANGE,
+        ROWVAL_SORTED_COLUMN,
+    )
+end
+
+
+"""
+    checkvalid(Bool, S::SparseMatrixCSC; full = true) -> Bool
```

Comment:
Hmm, looking at the docs for `checkbounds`:

>  `checkbounds(Bool, A, I...)`

>  Return true if the specified indices I are in bounds for the given array A.
>  Subtypes of AbstractArray should specialize this method if they need to
>  provide custom bounds checking behaviors; however, in many cases one can
>  rely on A's indices and checkindex.

so this is consistent with that. In my opinion `::Type{Bool}` is more clear though. Thoughts?